shared/src/commonMain/kotlin/com/karumi/gallery/data/local/LocalPhotosDataSource.kt

```diff
@@ -0,0 +1,35 @@
+package com.karumi.gallery.data.local
+
+import com.karumi.gallery.data.local.LocalGalleryDb.photosQueries
+import com.karumi.gallery.domain.TTLCache
+import com.karumi.gallery.model.PhotoShot
+import com.karumi.gallery.model.Photos
+
+class LocalPhotosDataSource(private val cache: TTLCache) {
+
+  fun insert(photos: Photos) {
+    photos.map(::insert)
+  }
+
+  private fun insert(photos: PhotoShot) {
+    photosQueries.transaction {
```

Comment:
Do we need a transaction for a single insertion?

Reply:
Same for the deletion below

Reply:
the number of insertions isn't related to the transaction right?, I mean if some other thread, try to insert in the same time, thanks to use transaction will be done in the right way if I am not wrong, it's not about things you are doing inside of a transaction is related to the possible calls from outside.